fastly/block_fastly_service_logging_newrelicotlp.go

```diff
@@ -0,0 +1,264 @@
+			Default:     "US",
+			Description: "The region that log data will be sent to. Default: `US`",
+		},
+		"token": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Sensitive:   true,
+			Description: "The Insert API key from the Account page of your New Relic account",
+		},
+	}
+
+	if h.GetServiceMetadata().serviceType == ServiceTypeVCL {
+		blockAttributes["format"] = &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Apache style log formatting. Your log must produce valid JSON that New Relic OTLP can ingest.",
+		}
+		blockAttributes["format_version"] = &schema.Schema{
+			Type:             schema.TypeInt,
+			Optional:         true,
+			Default:          2,
+			Description:      "The version of the custom logging format used for the configured endpoint. Can be either `1` or `2`. (default: `2`).",
+			ValidateDiagFunc: validateLoggingFormatVersion(),
+		}
+		blockAttributes["placement"] = &schema.Schema{
+			Type:             schema.TypeString,
+			Optional:         true,
+			Description:      "Where in the generated VCL the logging call should be placed.",
+			ValidateDiagFunc: validateLoggingPlacement(),
+		}
+		blockAttributes["response_condition"] = &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The name of the condition to apply.",
+		}
+		blockAttributes["url"] = &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The optional New Relic Trace Observer URL to stream logs to for Infinite Tracing.",
+		}
+	}
+
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: blockAttributes,
+		},
+	}
+}
+
+// Create creates the resource.
+func (h *NewRelicOTLPServiceAttributeHandler) Create(_ context.Context, d *schema.ResourceData, resource map[string]any, serviceVersion int, conn *gofastly.Client) error {
+	opts := h.buildCreate(resource, d.Id(), serviceVersion)
+
+	log.Printf("[DEBUG] Fastly New Relic OTLP logging addition opts: %#v", opts)
+
+	return createNewRelicOTLP(conn, opts)
+}
+
+// Read refreshes the resource.
+func (h *NewRelicOTLPServiceAttributeHandler) Read(_ context.Context, d *schema.ResourceData, _ map[string]any, serviceVersion int, conn *gofastly.Client) error {
+	localState := d.Get(h.GetKey()).(*schema.Set).List()
+
+	if len(localState) > 0 || d.Get("imported").(bool) || d.Get("force_refresh").(bool) {
+		log.Printf("[DEBUG] Refreshing New Relic OTLP logging endpoints for (%s)", d.Id())
+		remoteState, err := conn.ListNewRelicOTLP(&gofastly.ListNewRelicOTLPInput{
+			ServiceID:      d.Id(),
+			ServiceVersion: serviceVersion,
+		})
+		if err != nil {
+			return fmt.Errorf("error looking up New Relic OTLP logging endpoints for (%s), version (%v): %s", d.Id(), serviceVersion, err)
+		}
+
+		dll := flattenNewRelicOTLP(remoteState)
+
+		for _, element := range dll {
+			h.pruneVCLLoggingAttributes(element)
+		}
```

Comment:
We should remove this as the `pruneVCLLoggingAttribute()` method only works on Compute services...

https://github.com/fastly/terraform-provider-fastly/blob/68c8b8f9aca351d148dfd09fda9c46b2925903aa/fastly/service_attribute_definition.go#L95-L103